Avoid building problematic tests on platforms
whose linker does not support --gc-sections.

https://bugs.ntp.org/show_bug.cgi?id=3601
https://bugs.gentoo.org/564018

--- a/sntp/m4/ntp_problemtests.m4	2019-09-23 07:19:18.545861137 +0200
+++ b/sntp/m4/ntp_problemtests.m4	2019-09-23 07:19:18.545861137 +0200
@@ -33,6 +33,10 @@
  no:0:*-*-solaris*) ;;
  *) ntp_test_ntp_restrict="yes" ;;
 esac
+case "$ntp_cv_gc_sections_runs" in
+  no) ntp_test_ntp_restrict="no" ;;
+  * ) ;;
+esac
 AC_MSG_RESULT([$ntp_test_ntp_restrict])
 AM_CONDITIONAL([BUILD_TEST_NTP_RESTRICT], [test x$ntp_test_ntp_restrict = xyes])
 
@@ -43,6 +47,10 @@
  no:0:*-*-solaris*) ;;
  *) ntp_test_ntp_scanner="yes" ;;
 esac
+case "$ntp_cv_gc_sections_runs" in
+  no) ntp_test_ntp_scanner="no" ;;
+  * ) ;;
+esac
 AC_MSG_RESULT([$ntp_test_ntp_scanner])
 AM_CONDITIONAL([BUILD_TEST_NTP_SCANNER], [test x$ntp_test_ntp_scanner = xyes])
 
@@ -53,6 +61,10 @@
  no:0:*-*-solaris*) ;;
  *) ntp_test_ntp_signd="yes" ;;
 esac
+case "$ntp_cv_gc_sections_runs" in
+  no) ntp_test_ntp_signd="no" ;;
+  * ) ;;
+esac
 AC_MSG_RESULT([$ntp_test_ntp_signd])
 AM_CONDITIONAL([BUILD_TEST_NTP_SIGND], [test x$ntp_test_ntp_signd = xyes])
 
